### ISSUE TITLE
Biber doesn't sort by entrytype

### ISSUE THREAD

**Issue by moewew** [plk/biber#206]
Consider

```
\documentclass{article}

\usepackage[backend=biber]{biblatex}
\addbibresource{biblatex-examples.bib}
\DeclareSortingTemplate{entrytype}{
  \sort{\field{entrytype}}
}

\begin{document}
\nocite{cicero,sigfridsson}

\begin{refcontext}[sorting=entrytype]
\printbibliography
\end{refcontext}
\end{document}
```
this sorts `cicero` > `sigfridsson` even though sorting by entrytype should give `sigfridsson` (`@article`) > `cicero` (`@book`)
The `.bbl` contains
```
[651] Internals.pm:1102> DEBUG - Sorting object for key 'cicero' -> ["entrytype"]
[651] Internals.pm:1102> DEBUG - Sorting object for key 'sigfridsson' -> ["entrytype"]
```
but of course we expected the values `book` and `article`.

This came up in https://tex.stackexchange.com/a/346895/35864

**Comment by moewew** [comment]
I suppose 2.10 and `biblatex` 3.10 have already been sent off to CTAN? So this came in just a few hours to late....

**Comment by plk** [comment]
No not yet as I’m waiting for the biber builders to commit so I should be able to fix this

**Comment by plk** [comment]
Fixed in v2.10 - this will make to the CTAN release.

**Comment by moewew** [comment]
Works brilliantly with the version from sourceforge. Thank you very much.
